tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs

```diff
@@ -0,0 +1,108 @@
+            string userText = (previewedCard.Body[1] as AdaptiveTextBlock).Text;
+            var choiceSet = previewedCard.Body[3] as AdaptiveChoiceSetInput;
+
+            return new SubmitExampleData()
+            {
+                Question = userText,
+                MultiSelect = choiceSet.IsMultiSelect ? "true" : "false",
+                Option1 = choiceSet.Choices[0].Title,
+                Option2 = choiceSet.Choices[1].Title,
+                Option3 = choiceSet.Choices[2].Title,
+            };
+        }
+
+        public static MessagingExtensionActionResponse GetTaskModuleAdaptiveCardResponse(string userText = null, bool isMultiSelect = true, string option1 = null, string option2 = null, string option3 = null)
+        {
+            return new AdaptiveCard()
+            {
+                Body = new List<AdaptiveElement>()
+                {
+                    new AdaptiveTextBlock("This is an Adaptive Card within a Task Module")
+                    {
+                        Weight = AdaptiveTextWeight.Bolder,
+                    },
+                    new AdaptiveTextBlock("Enter text for Question:"),
+                    new AdaptiveTextInput() { Id = "Question", Placeholder = "Question text here", Value = userText },
+                    new AdaptiveTextBlock("Options for Question:"),
+                    new AdaptiveTextBlock("Is Multi-Select:"),
+                    new AdaptiveChoiceSetInput()
+                    {
+                        Type = AdaptiveChoiceSetInput.TypeName,
+                        Id = "MultiSelect",
+                        Value = isMultiSelect ? "true" : "false",
+                        IsMultiSelect = false,
+                        Choices = new List<AdaptiveChoice>()
+                        {
+                            new AdaptiveChoice() { Title = "True", Value = "true" },
+                            new AdaptiveChoice() { Title = "False", Value = "false" },
+                        },
+                    },
+                    new AdaptiveTextInput() { Id = "Option1", Placeholder = "Option 1 here", Value = option1 },
+                    new AdaptiveTextInput() { Id = "Option2", Placeholder = "Option 2 here", Value = option2 },
+                    new AdaptiveTextInput() { Id = "Option3", Placeholder = "Option 3 here", Value = option3 },
+                },
+                Actions = new List<AdaptiveAction>()
+                {
+                    new AdaptiveSubmitAction
+                    {
+                        Type = AdaptiveSubmitAction.TypeName,
+                        Title = "Submit",
+                        Data = new JObject { { "submitLocation", "messagingExtensionFetchTask" } },
+                    },
+                },
+            }.ToTaskModuleResponse();
+        }
+
+        public static AdaptiveCard GetCardFromUserSubmittedData(SubmitExampleData data)
```

Comment:
Again odd function name?
Possible candidate for an extension function? for example, along the lines submitExampleData.ToAdaptiveCard()

Reply:
Change to ToAdaptiveCard